lib/rubocop/cop/sorbet/obsolete_strict_memoization.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+require "rubocop"
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # This cop checks for the obsolete pattern for initializing instance variables that was required for older Sorbet
+      # versions in `#typed: strict` files.
+      #
+      # It's no longer required, as of Sorbet 0.5.10210
+      # See https://github.com/sorbet/sorbet/issues/995
+      #
+      # @example
+      #
+      #   # bad
+      #   sig { returns(Foo) }
+      #   def foo
+      #     @foo = T.let(@foo, T.nilable(Foo))
+      #     @foo ||= Foo.new
+      #   end
+      #
+      #   # good
+      #   sig { returns(Foo) }
+      #   def foo
+      #     @foo ||= T.let(Foo.new, T.nilable(Foo))
+      #   end
+      #
+      # TODO: find the right way to access the line length limit, indentation style, and indentation width.
+      class ObsoleteStrictMemoization < RuboCop::Cop::Base
+        include RuboCop::Cop::MatchRange
+        extend AutoCorrector
+
+        include TargetSorbetVersion
+        minimum_target_sorbet_static_version "0.5.10210"
+
+        MESSAGE = "This two-stage workaround for memoization in `#typed: strict` files is no longer necessary. " \
+          "See https://github.com/sorbet/sorbet/issues/995."
+
+        # @!method legacy_memoization_pattern?(node)
+        def_node_matcher :legacy_memoization_pattern?, <<~PATTERN
+          (begin
+            (ivasgn $_ivar                                # @_ivar = ...
+              (send                                       # T.let(_ivar, T.nilable(_ivar_type))
+                (const nil? :T) :let
+                (ivar _ivar)
```

Comment:
Interesting idea... do you mean this?

```ruby
sig { returns(Foo) }
def foo
  @foo = T.let(nil, T.nilable(Foo))
  #            ^^^
  @foo ||= Foo.new
end
```

That's more of a misapplication of the pattern... seems important to check, since it would disable the memoization behaviour entirely haha

Reply:
I counted very approximatively how many of these cases we have in core (`git grep -A 1 "@.* = T.let(nil," | grep -B 1 ".*\.new"`) and it seems it's only a handful, maybe we can fix or disregard them 👍

Reply:
Implemented!